arxivdigest/api/search.py

```diff
@@ -1,22 +1,33 @@
 """Search functionality using Elasticsearch."""
 
-from elasticsearch import Elasticsearch
-
-es = Elasticsearch(['http://localhost:9200'])
-
-def search_articles(query, page=1, per_page=10):
-    """Search articles using Elasticsearch."""
+def search_articles(es, query, page=1, per_page=10):
+    """Search articles using Elasticsearch.
+    
+    Args:
+        es: Elasticsearch instance
+        query: Search query string
+        page: Page number (default: 1)
+        per_page: Results per page (default: 10)
+    """
     from_index = (page - 1) * per_page
     
+    # Get index from config
+    import json
+    import os
+    config_path = os.path.join(os.path.dirname(__file__), '..', '..', 'config.json')
+    with open(config_path) as f:
+        config = json.load(f)
+    index = config.get('elasticsearch_config', {}).get('index', 'arxiv')
```

Comment:
The config is loaded in core/config.py and can be at different locations. Load `elastic_index_name` there.